pennylane_lightning/lightning_tensor/_state_tensor.py

```diff
@@ -0,0 +1,187 @@
+# Copyright 2024 Xanadu Quantum Technologies Inc.
+
+# Licensed under the Apache License, Version 2.0 (the "License");
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+
+#     http://www.apache.org/licenses/LICENSE-2.0
+
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+"""
+Class implementation for state-tensor manipulation.
+"""
+
+# pylint: disable=import-error, no-name-in-module, ungrouped-imports
+try:
+    from pennylane_lightning.lightning_tensor_ops import StateTensorC64, StateTensorC128
+except ImportError:
+    pass
+
+
+import numpy as np
+import pennylane as qml
+from pennylane import BasisState, DeviceError, StatePrep
+from pennylane.ops.op_math import Adjoint
+from pennylane.tape import QuantumScript
+from pennylane.wires import Wires
+
+
+class LightningStateTensor:
+    """Lightning state-tensor class.
+
+    Interfaces with C++ python binding methods for state-tensor manipulation.
+
+    Args:
+        num_wires(int): the number of wires to initialize the device with
+        max_bond_dim(int): maximum bond dimension for the state tensor
+        dtype: Datatypes for state-tensor representation. Must be one of
+            ``np.complex64`` or ``np.complex128``. Default is ``np.complex128``
+        device_name(string): state tensor device name. Options: ["lightning.tensor"]
+    """
+
+    def __init__(
+        self, num_wires, max_bond_dim, dtype=np.complex128, device_name="lightning.tensor"
+    ):
+        self._num_wires = num_wires
+        self._wires = Wires(range(num_wires))
```

Comment:
If `num_wires` is passed in as a range what will happen here? Is it handled at a higher layer, so we don't need to worry?

Reply:
Good catch, yea, we don't need to worry it for now. Removed here.